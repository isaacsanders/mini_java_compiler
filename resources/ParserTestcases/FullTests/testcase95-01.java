class Test {
    public static void main(String[] args) {
        boolean a = true;
        boolean b = true;
        if ((!a && b) || (a && !b)) {
            if ( a ) {
                a = !a;
            } else {
                b = !b;
            }
        } else {
            int i = 1321;
        }
    }    
}