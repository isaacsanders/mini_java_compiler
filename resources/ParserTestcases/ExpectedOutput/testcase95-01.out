StmtList ::= 
Type ::= BOOLEAN
LiteralExpr ::= TRUE
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Type ::= BOOLEAN
LiteralExpr ::= TRUE
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= NOT UnaryExpr
Expr ::= UnaryExpr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr AND Expr
ParenExpr	::=	LPAREN Expr RPAREN
LiteralExpr ::= ParenExpr
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= NOT UnaryExpr
Expr ::= UnaryExpr
Expr ::= Expr AND Expr
ParenExpr	::=	LPAREN Expr RPAREN
LiteralExpr ::= ParenExpr
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr OR Expr
StmtList ::= 
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
StmtList ::= 
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= NOT UnaryExpr
Expr ::= UnaryExpr
Stmt ::= ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Stmt ::= LBRACE StmtList RBRACE
StmtList ::= 
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= NOT UnaryExpr
Expr ::= UnaryExpr
Stmt ::= ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Stmt ::= LBRACE StmtList RBRACE
Stmt ::= IF LPAREN Expr RPAREN Stmt ELSE Stmt
StmtList ::= StmtList Stmt
Stmt ::= LBRACE StmtList RBRACE
StmtList ::= 
Type ::= INT
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Stmt ::= LBRACE StmtList RBRACE
Stmt ::= IF LPAREN Expr RPAREN Stmt ELSE Stmt
StmtList ::= StmtList Stmt
MainClassDecl ::= ...
ClassDeclList ::= 
Program ::= MainClassDecl ClassDeclList
